analyzers/tests/SonarAnalyzer.UnitTest/Rules/StringLiteralShouldNotBeDuplicatedTest.cs

```diff
@@ -28,30 +28,41 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class StringLiteralShouldNotBeDuplicatedTest
     {
+        private readonly VerifierBuilder csBuilder = new VerifierBuilder<CS.StringLiteralShouldNotBeDuplicated>();
         [TestMethod]
         public void StringLiteralShouldNotBeDuplicated_CS() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\StringLiteralShouldNotBeDuplicated.cs", new CS.StringLiteralShouldNotBeDuplicated());
+            csBuilder.AddPaths("StringLiteralShouldNotBeDuplicated.cs").Verify();
 
 #if NET
         [TestMethod]
         public void StringLiteralShouldNotBeDuplicated_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Console(@"TestCases\StringLiteralShouldNotBeDuplicated.CSharp9.cs", new CS.StringLiteralShouldNotBeDuplicated());
+            csBuilder.AddPaths("StringLiteralShouldNotBeDuplicated.CSharp9.cs").WithOptions(ParseOptionsHelper.FromCSharp9).Verify();
```

Comment:
Because `CompilationUnit` is used as an entry point it finds positives no matter what `INamedType` is. Therefore I decided to create an extra test file to make sure, the analyzer operates as intended in "normal" and in "top-level statement" context. As I said in the PR description, `record structs` seemed to be already supported but that was only in the top-level statement case.
In general, I found it useful to have a test file dedicated to top-level statements only in the other PRs.